lib/msf/core/db_manager/http/servlet/module_servlet.rb

```diff
@@ -0,0 +1,28 @@
+module ModuleServlet
+
+  def self.api_path
+    '/api/v1/modules'
+  end
+
+  def self.registered(app)
+    app.get ModuleServlet.api_path, &search_modules
+  end
+
+  #######
+  private
+  #######
+
+  def self.search_modules
+    lambda {
+      begin
```

Comment:
Between the lambda and begin there should be a call to `warden.authenticate!` such that requests are authenticated. The REST API authentication PR landed yesterday so those changes are available in upstream-master.